custom_components/elasticsearch/config_flow.py

```diff
@@ -441,6 +442,21 @@ async def _async_elasticsearch_login(self) -> ClusterCheckResult:
             await privilege_check.enforce_privileges(self.config)
 
             version = gateway.es_version
+        except ClientError:
+            # For a Client Error, try to initialize without SSL verification, if this works then there is a self-signed certificate being used
+            gateway = ElasticsearchGateway(raw_config=self.config)
+            gateway._verify_certs = False
+            try:
+                await gateway.async_init()
+
+                privilege_check = ESPrivilegeCheck(gateway, config=self.config)
+                await privilege_check.enforce_privileges(self.config)
```

Comment:
By re-using the `gateway` variable, we are preventing the original created above from gracefully shutting down.

We should either use different variable names, which we cleanup in their own `finally` block, -or- attempt to shutdown the first `gateway` before reusing the variable on this new instance.